### PR TITLE
Updates from user testing and core-toggle aria-controls

### LINKED ISSUE #118: Feedback from user-testing
- [x] core-toggle: should support moving focus to element with `autofocus` attribute on open
- [x] core-datepicker: remove `aria-disabled`
- [x] core-datepicker: add full date to `aria-label` on date-table
- [x] core-datepicker: add `autofocus` on selected date in date-table
- [x] core-datepicker: add documentation on why writing days like mon/thu/web is better than m/t/w 
- [x] core-dialog: do not close if `event.defaultPrevented` is set on ESC
- [x] core-dialog: add documentation on how markup position affects usability (should be after button)
- [x] core-dialog: add documentation on `autofocus` attribute
- [x] core-input: add documentation on why "searching" and "no hits" is important to implement when using `ajax`
- [x] core-input: move `hidden` attribute to `<li>` instead of `<button>`

### PR DESCRIPTION
**Pull request by eirikbacker**
- Fixes #118 and #117

### CODE REVIEW

File: packages/core-dialog/core-dialog.js [hunk 1]
```diff
@@ -130,6 +128,7 @@ function queryFocusable (context) {
 }
 
 function setFocus (dialog) {
+  if (dialog.contains(document.activeElement)) return // Do not move if focus is allready inside
```
Comment (wwalmnes) [inline packages/core-dialog/core-dialog.js]: Allready -> already

File: packages/core-dialog/readme.md [hunk 2]
```diff
@@ -223,12 +222,24 @@ function closeDialog (event) {
...
 <!--[if IE 9]><script>document.createElement('dialog')</script><![endif]-->
 ```
 
+### Elements order
+
+Though not strictly required, the `<button>` opening a `@nrk/core-dialog` should be placed directly before the `<dialog>` itself. This eases the mental model for screen reader users.
+
+### Autofocus
+
+If you have form elements inside a `@nrk/core-dialog`, you can optionally add a `autofocus` attribute to the most prominent form element. This helps the user navigate quickly when toggle is opened.
```
Comment (wwalmnes) [inline packages/core-dialog/readme.md]: add a autofocus --> add an autofocus

File: packages/core-input/core-input.js [hunk 3]
```diff
@@ -91,8 +92,11 @@ function onSelect (input, detail) {
 function onFilter (input, detail) {
   if (dispatchEvent(input, 'input.filter', detail) && !ajax(input)) {
     queryAll(ITEM, input.nextElementSibling).reduce((acc, item) => {
+      const list = item.parentElement.nodeName === 'LI' && item.parentElement
       const show = item.textContent.toLowerCase().indexOf(input.value.toLowerCase()) !== -1
-      item[show ? 'removeAttribute' : 'setAttribute']('hidden', '')
+      const attr = show ? 'removeAttribute' : 'setAttribute'
+      if (list) list[attr]('hidden', '') // JAWS requires hiding of <li> too (if they exsist)
```
Comment (wwalmnes) [inline packages/core-input/core-input.js]: exsist --> exist

**Comment by eirikbacker** [comment]
Thanks for review @wwalmnes :) Have fixed typos, and also added a `input.ajax.beforeSend` event so #114 can be resolved - feedback much appreciated ❤️

File: packages/core-input/core-input.js [hunk 4]
```diff
@@ -14,6 +14,8 @@ export default function input (elements, content) {
     const list = input.nextElementSibling
     const ajax = typeof options.ajax === 'undefined' ? input.getAttribute(UUID) : options.ajax
 
+
+
```
Comment (wwalmnes) [inline packages/core-input/core-input.js]: No need for extra lines

File: packages/core-input/core-input.js [hunk 5]
```diff
@@ -126,10 +128,12 @@ function ajax (input) {
...
     dispatchEvent(input, 'input.ajax', req)
   }
-  ajax.timer = setTimeout(() => { // Debounce next request 500 milliseconds
+  ajax.timer = setTimeout(() => {
     if (!input.value) return // Abort if input is empty
-    req.open('GET', url.replace('{{value}}', window.encodeURIComponent(input.value)), true)
-    req.setRequestHeader('X-Requested-With', 'XMLHttpRequest') // https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Requested-With
-    req.send()
-  }, AJAX_DEBOUNCE)
+    if (dispatchEvent(input, 'input.ajax.beforeSend', req)) {
```
Comment (wwalmnes) [inline packages/core-input/core-input.js]: Maybe a stupid question, but what if in the `ajax.beforeSend` event handler the user executes `req.send()`/`req.open()`? Will that trigger an error when we call `req.send()`/`req.open()`?  Should we handle that?
